nixos/modules/services/web-apps/firefly-iii.nix

```
@@ -0,0 +1,356 @@
+
+    dataDir = mkOption {
+      description = mdDoc "Firefly III data directory";
+      default = "/var/lib/firefly-iii";
+      type = path;
+    };
+
+    user = mkOption {
+      type = str;
+      default = defaultUser;
+      description = mdDoc "User account under which firefly-iii runs.";
+    };
+
+    group = mkOption {
+      type = str;
+      default = defaultGroup;
+      description = mdDoc ''
+        Group under which firefly-iii runs. It is best to set this to the group
+        of whatever webserver is being used as the frontend.
+      '';
+    };
+
+    package = mkOption {
+      type = package;
+      default = firefly-pkg;
+      defaultText = literalExpression "pkgs.firefly-iii";
+      description = mdDoc ''
+        The firefly-iii package served by php-fpm and the webserver of choice.
+        This option can be used to point the webserver to the correct root. It
+        may also be used to set the package to a different version, say a
+        development version.
+      '';
+    };
+
+    # PHP
+    poolConfig = mkOption {
+      type = attrsOf (oneOf [ str int bool ]);
+      default = {
+        "pm" = "dynamic";
+        "pm.max_children" = 32;
+        "pm.start_servers" = 2;
+        "pm.min_spare_servers" = 2;
+        "pm.max_spare_servers" = 4;
+        "pm.max_requests" = 500;
+      };
+      description = mdDoc ''
+        Options for the Firefly III PHP pool. See the documentation on <literal>php-fpm.conf</literal>
+        for details on configuration directives.
+      '';
+    };
+
+    settings = mkOption {
+      default = {};
+      description = mdDoc ''
+        Options for firefly-iii configuration. Refer to
+        <https://github.com/firefly-iii/firefly-iii/blob/main/.env.example> for
+        details on supported values. All <option>_FILE values supported by
+        upstream are supported here.
+      '';
+      example = literalExpression ''
+        {
+          APP_ENV = "production";
+          APP_URL = "https://example.com";
+          APP_KEY = "SomeKeyOfExactly32Chars";
+          SITE_OWNER = "mail@example.com";
+          DB_CONNECTION = "mysql";
+          DB_HOST = "db";
+          DB_PORT = 3306;
+          DB_DATABASE = "firefly";
+          DB_USERNAME = "firefly";
+          DB_PASSWORD_FILE = "/var/secrets/firefly-iii-mysql-password.txt;
+        }
+      '';
+      type = submodule {
+        freeformType = format.type;
+        options = {
+          APP_ENV = mkOption {
+            default = "local";
+            type = enum [ "production" "local" "testing" ];
+            description = mdDoc ''
+              Top level option governing Firefly's behaviour. This is best left
+              at "local".
+            '';
+          };
+          APP_URL = mkOption {
+            description = mdDoc ''
+              The root URL that you want to host Firefly III on. All URLs
+              in Firefly III will be generated using this value.
```

```suggestion
              The root URL that you want to host Firefly III on.
              All URLs in Firefly III will be generated using this value.
```
Lets make this easy to read for everyone